File: easy/1446_consecutive_characters/solution.cpp
#include <string>

class Solution {
public:
  int maxPower(std::string s) {
    int len = 1;
    int maxLen = 1;

    for (size_t i = 0; i < s.size() - 1; ++i) {
      if (s[i] == s[i + 1])
        ++len;
      else {
        maxLen = std::max(maxLen, len);
        len = 1;
      }
    }

    return std::max(maxLen, len);
  }
};

File: easy/1446_consecutive_characters/solution.py
class Solution:
    def maxPower(self, s: str) -> int:
        power = 1
        max_power = 1

        for i in range(len(s) - 1):
            if s[i] == s[i + 1]:
                power += 1
            else:
                max_power = max(max_power, power)
                power = 1

        return max(max_power, power)
